src/fp6.rs

```diff
@@ -681,3 +687,74 @@ fn test_zeroize() {
     a.zeroize();
     assert!(bool::from(a.is_zero()));
 }
+
+#[cfg(test)]
+mod tests {
+    use rand_core::SeedableRng;
+
+    use super::*;
+    const TEST_ITER: usize = 1000;
+    const SEED: [u8; 16] = [
+        0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc,
+        0xe5,
+    ];
+
+    // this is meant to mirror the zkvm branch of mul_interleaved
+    fn mul_interleaved_zkvm_test(a: Fp6, b: &Fp6) -> Fp6 {
+        // Implements Algorithm 13 from https://eprint.iacr.org/2010/354.pdf
+        let mut t0 = a.c0;
+        t0 = t0.mul(&b.c0);
+        let mut t1 = a.c1;
+        t1 = t1.mul(&b.c1);
+        let mut t2 = a.c2;
+        t2 = t2.mul(&b.c2);
+        let mut c0 = a.c1;
+        c0 = c0.add(&a.c2);
+        let mut tmp = b.c1;
+        tmp = tmp.add(&b.c2);
+        c0 = c0.mul(&tmp);
+        tmp = t2;
+        tmp = tmp.add(&t1);
+        c0 = c0.sub(&tmp);
+        c0 = c0.mul_by_nonresidue();
+        c0 = c0.add(&t0);
+        let mut c1 = a.c0;
+        c1 = c1.add(&a.c1);
+        tmp = b.c0;
+        tmp = tmp.add(&b.c1);
+        c1 = c1.mul(&tmp);
+        tmp = t0;
+        tmp = tmp.add(&t1);
+        c1 = c1.sub(&tmp);
+        tmp = t2.mul_by_nonresidue();
+        c1 = c1.add(&tmp);
+        tmp = a.c0;
+        tmp = tmp.add(&a.c2);
+        let mut c2 = b.c0;
+        c2 = c2.add(&b.c2);
+        c2 = c2.mul(&tmp);
+        tmp = t0;
+        tmp = tmp.add(&t2);
+        c2 = c2.sub(&tmp);
+        c2 = c2.add(&t1);
+        Fp6 { c0, c1, c2 }
+    }
+
+    #[test]
+    fn fuzz_mul_interleaved() {
+        for _i in 0..TEST_ITER {
+            let mut rng = rand_xorshift::XorShiftRng::from_seed(SEED);
+            let a = Fp6::random(&mut rng);
+            let b = Fp6::random(&mut rng);
```

Comment:
```suggestion
        let mut rng = rand_xorshift::XorShiftRng::from_seed(SEED);
        for _i in 0..TEST_ITER {
            let a = Fp6::random(&mut rng);
            let b = Fp6::random(&mut rng);
```
This ensures every test iteration is actually different

Reply:
Oh my!